src/others/helpers/map-utils.ts

```diff
@@ -1,160 +1,51 @@
-import { groupBy } from "lodash";
-import omit from "lodash/omit";
 import { AidRequest, Supply, Location } from "../contexts/api";
-import { DecodedAidRequest, DecodedAidRequestGroupedByLocation } from "./decode-aid-request";
 import type { Feature, Geometry, GeoJsonProperties } from "geojson";
 
-type GroupedByDate = {
-  date: string;
-  aidRequests: Omit<DecodedAidRequest, "date">[]; // TODO: consider replacing with AidRequest
-};
-
-type GroupedByDateWithTotal = {
-  date: string;
-  aidRequests: Omit<DecodedAidRequest, "date">[]; // TODO: consider replacing with AidRequest
+type GroupedByCityId = {
+  city_id: number;
   total: number;
-  totalDescription: string;
-};
-
-type GroupedByLocationAndDateWithTotal = {
-  location: Location;
-  aidRequestsByDateWithTotal: GroupedByDateWithTotal[];
+  aidRequests: AidRequest[];
 };
 
-interface RequestProperties {
+type MapData = {
+  city_id: number;
   amount: number;
-  normalized_amount?: number;
-  date: string;
-  category: string;
   description: string;
-  city: string;
-}
-
-type GroupedByCityId = {
-  city_id: number;
-  total: number;
-  aidRequests: AidRequest[];
 };
 
-// TODO: Delete once map relies on AidRequest data model
-export const adaptToMap = (
-  aidRequestGroupedByCityId: GroupedByCityId,
-  location: Location | undefined,
+export const aggregateCategories = (
```

Comment:
**issue**

Not the best naming, imo. I don't think we aggregate here anything. If the input is aidRequest and output is a MapDataPoint, and the function does mapping, I think the naming should reflect this: sth like: `mapAidRequestToMapDataPoint`?

I purposefully suggest naming in singular, because for clarity it should take single data item, not the array. That would refrain developers from putting any kind of additional logic into the body of the mapper, like sorting, which is not its responsibility. Its sole responsibility should be to map one entity into another.

Reply:
1) re "mapAidRequestToMapDataPoint" we are not mapping one AidRequest here, but a group of AidRequests for a given city, so does not seem like "map" is appropriate
2) re "don't think we aggregate here anything" - we do aggregate all selected categories here into just one record
3) re "sort" we need a sort here, so that items in the description are properly ordered

Reply:
re.1: if input is an array of length n and autput is an array of length n, then you do map. 
re1: but imo it will be more useful to have 1:1 mapping here

re2: yes, but you aggregate the data for each entity, not the entities itself - which is counfusing; if it wasn't so, I would not comment :)

re3. as I mentioned earlier, sorting can be done separately, doesn't need to be part of this function.